include/seqan3/alignment/pairwise/alignment_configurator.hpp

```diff
@@ -157,33 +165,196 @@ struct alignment_configurator
+     * \tparam config_t The alignment configuration type.
+     * \param[in] cfg   The passed configuration object.
+     */
+    template <typename kernel_t, typename config_t>
+    static constexpr auto configure_free_ends(config_t const &);
+
+    /*!\brief Configures the find_optimum_policy for the alignment algorithm.
+     * \tparam kernel_t     The type-erased kernel to return.
+     * \tparam gap_policy_t The type of the core alignment policy needed for the find_optimum_policy.
+     * \tparam policies_t   A template parameter pack for the remaining policies.
+     * \tparam config_t     The alignment configuration type.
+     * \param[in] cfg       The passed configuration object.
+     *
+     * \details
+     *
+     * The find optimum policy is used to track the optimal score and position if requested.
+     * Some algorithms such as local alignments require to test every cell in the matrix for their optimum.
+     * Thus the core alignment policy computing the cells must be able to access the functions in the
+     * find_optimum_policy. Since both classes are crtp base classes of the alignment algorithm the find_optimum_policy
+     * must befriend the core_policy_t in order to grant it access to the contained member functions.
+     */
+    template <typename kernel_t, typename core_policy_t, typename ...policies_t, typename config_t>
+    static constexpr auto configure_find_optimum(config_t const &);
+
 };
 
+//!\cond
+template <typename kernel_t, typename config_t>
+constexpr auto
+alignment_configurator::configure_free_ends(config_t const & cfg)
+{
+    // ----------------------------------------------------------------------------
+    // score and cell type
+    // ----------------------------------------------------------------------------
+
+    using score_type = int;
+    using cell_type = std::tuple<score_type, score_type>;
+
+    // ----------------------------------------------------------------------------
+    // dynamic programming matrix
+    // ----------------------------------------------------------------------------
+
+    using dp_matrix_t = deferred_crtp_base<unbanded_dp_matrix_policy, std::allocator<cell_type>>;
+
+    // ----------------------------------------------------------------------------
+    // affine gap kernel
+    // ----------------------------------------------------------------------------
+
+    using affine_t = deferred_crtp_base<affine_gap_policy, cell_type>;
+
+    // ----------------------------------------------------------------------------
+    // configure initialization policy
+    // ----------------------------------------------------------------------------
+
+    // Get the value for the sequence ends configuration.
+    auto align_ends_cfg = cfg.template value_or<align_cfg::aligned_ends>(align_cfg::none_ends_free);
+    using align_ends_cfg_t = decltype(align_ends_cfg);
+
+    // This lambda augments the initialization policy of the alignment algorithm
```

Comment:
```suggestion
    // This lambda augments the initialisation policy of the alignment algorithm
```

Also everywhere else if accepted